src/System.Management.Automation/engine/hostifaces/Command.cs

```diff
@@ -497,13 +509,27 @@ CommandOrigin origin
                     commandProcessorBase = new CommandProcessor(functionInfo, executionContext,
                                                                 _useLocalScope ?? false, fromScriptFile: false, sessionState: executionContext.EngineSessionState);
                 }
+                else if (ScriptPath != null)
+                {
+                    var scriptName = Path.GetFileName(ScriptPath);
```

Comment:
The script (content of the file) could be using cmdlet binding (`scriptBlock.UsesCmdletBinding` being `true`), by having a param block defined. In that case, the code execution would fall into the `if (scriptBlock.UsesCmdletBinding)` block, right?